modules/solace/solace.go

```diff
@@ -0,0 +1,291 @@
+)
+
+type SolaceContainer struct {
+	Container      testcontainers.Container
+	Username       string
+	Password       string
+	Vpn            string
+	queues         map[string][]string // queueName -> topics
+	WithClientCert bool
+	Image          string
+	ctx            context.Context
+	exposedPorts   []string
+	envVars        map[string]string
+	shmSize        int64
+}
+
+// WithCredentials sets the client credentials (username, password)
+func (s *SolaceContainer) WithCredentials(username, password string) *SolaceContainer {
+	s.Username = username
+	s.Password = password
+	return s
+}
+
+// WithVpn sets the VPN name
+func (s *SolaceContainer) WithVpn(vpn string) *SolaceContainer {
+	s.Vpn = vpn
+	return s
+}
+
+// WithQueue subscribes a given topic to a queue (for SMF/AMQP testing)
+func (s *SolaceContainer) WithQueue(queueName, topic string) *SolaceContainer {
+	// Store queue/topic mapping in a new field for later use (e.g., CLI script generation)
+	if s.queues == nil {
+		s.queues = make(map[string][]string)
+	}
+	s.queues[queueName] = append(s.queues[queueName], topic)
+	return s
+}
+
+// WithEnv allows adding or overriding environment variables
+func (s *SolaceContainer) WithEnv(env map[string]string) *SolaceContainer {
+	if s.envVars == nil {
+		s.envVars = make(map[string]string)
+	}
+	for k, v := range env {
+		s.envVars[k] = v
+	}
+	return s
+}
+
+// WithExposedPorts allows adding extra exposed ports
+func (s *SolaceContainer) WithExposedPorts(ports ...string) *SolaceContainer {
+	s.exposedPorts = append(s.exposedPorts, ports...)
+	return s
+}
+
+const (
+	DefaultImage = "solace/solace-pubsub-standard:latest"
+	DefaultVpn   = "default"
+	DefaultUser  = "root"
+	DefaultPass  = "password"
+)
+
+func NewSolaceContainer(ctx context.Context, image string) *SolaceContainer {
+	if image == "" {
+		image = DefaultImage
+	}
+	return &SolaceContainer{
+		Username: DefaultUser,
+		Password: DefaultPass,
+		Vpn:      DefaultVpn,
+		Image:    image,
+		ctx:      ctx,
+	}
+}
+
+func (s *SolaceContainer) WithShmSize(size int64) *SolaceContainer {
+	s.shmSize = size
+	return s
+}
+
+// waitForSolaceActive waits for the Solace broker to be fully active by checking the system log
+func (s *SolaceContainer) waitForSolaceActive(ctx context.Context) error {
+	const maxAttempts = 60
+	const intervalSeconds = 1
+
+	for attempt := 0; attempt < maxAttempts; attempt++ {
+		// Execute grep command to check for "Primary Virtual Router is now active" in system log
+		code, _, err := s.Container.Exec(ctx, []string{"grep", "-R", "Primary Virtual Router is now active", "/usr/sw/jail/logs/system.log"})
+
+		if err == nil && code == 0 {
+			// Found the message, broker is active
+			return nil
+		}
+
+		// Wait before next attempt
+		time.Sleep(intervalSeconds * time.Second)
+	}
+
+	return fmt.Errorf("timeout waiting for Solace broker to become active after %d seconds", maxAttempts)
+}
+
+func (s *SolaceContainer) Run(ctx context.Context) error {
```

Comment:
issue: this pattern breaks the consistency with the rest of the library modules. Please check my previous comment.

This function should initialise a container request, and apply the passed in options. Also not being attached to the container type but as a public function in the package.

Reply:
updated to align with the rest of the modules.